### PR TITLE
fix: deeplink handling issue when the app is closed

### LINKED ISSUE #8466: [Bug]: Deeplinks don't work: `Network with chain id 1 notr found in your wallet`
### Describe the bug

**Problem:** deeplinks are not working, whenever I trigger a send ETH in Mainnet I get the error `Network with chain id 1 notr found in your wallet`. However I do have Mainnet network in the wallet

### Expected behavior

Deeplinks should work

### Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/54408225/5b22b2c6-f19a-40e5-bbef-f6573e44f0b3

### Steps to reproduce

1. Generate a deeplink (fill the recipient, amount and chain id) https://metamask.github.io/metamask-deeplinks/#
2. Scan it
3. See error

### Error messages or log output

_No response_

### Version

7.16.0

### Build type

None

### Device

Pixel 6

### Operating system

Android

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by omridan159**
## **Description**

This PR addresses a deeplink navigation issue found in MetaMask Mobile v7.10.0 and above ([LINK TO THE BUG](https://github.com/MetaMask/metamask-mobile/issues/7766)). The update introduces key improvements for handling deeplinks:

**Issue Overview:**
Previously, deeplinks failed to function correctly when the MetaMask app was not running in the background. This was caused by premature route navigation attempts before the app completed its loading process.

**Implemented Solutions:**

**Conditional Deeplink Queueing**:
A queue system for deeplinks has been implemented. Deeplinks that are triggered by [branch.io](http://branch.io/) events are queued if the app is not fully initialized. This queued processing ensures that deeplinks are handled correctly only once the app loads completely.

**Immediate Handling When the App Loaded:**
For instances where the app is already loaded, deeplinks from [branch.io](http://branch.io/) events are processed immediately and normally, without queuing. This ensures direct and standard handling of deeplinks in an active app state.

## **Related issues**

Fixes https://github.com/MetaMask/metamask-mobile/issues/8466

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**

https://github.com/MetaMask/metamask-mobile/assets/61094771/629803d3-0dea-4dd8-95b0-c509d248b07a

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Nav/App/index.js [hunk 1]
```diff
@@ -345,11 +347,20 @@ const App = ({ userLoggedIn }) => {
...
+              queueOfHandleDeeplinkFunctions.concat(() => {
+                handleDeeplink(opts);
+              }),
+            );
+          }
         });
       }
       prevNavigator.current = navigator;
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (abretonc7s) [inline app/components/Nav/App/index.js:363]: need to remove the disable-line and replace with including dependency in list

File: app/components/Nav/App/index.js [hunk 2]
```diff
@@ -401,6 +413,7 @@ const App = ({ userLoggedIn }) => {
     handlePostInit().catch((err) => {
       Logger.error(err, 'Error postInit SDKConnect');
     });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (abretonc7s) [inline app/components/Nav/App/index.js:416]: same here, include dependency in list

**Review by abretonc7s — CHANGES_REQUESTED** [review]
useEffect dependencies

**Review by abretonc7s — DISMISSED** [review]
LGTM

**Review by deeeed — APPROVED** [review]
LGTM

**Comment by christopherferreira9** [comment]
Android:

https://github.com/MetaMask/metamask-mobile/assets/104831203/5038dafe-c300-486e-a380-1ba354140598

iOS:

https://github.com/MetaMask/metamask-mobile/assets/104831203/5961def8-6c2f-42d8-a4ae-f13f3d4a32d4

Works as expected now.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**